plaid/plaid.go

```diff
@@ -1,373 +1,147 @@
-// Package plaid implements a Go client for the Plaid API (https://plaid.com/docs)
 package plaid
 
 import (
+	"bytes"
 	"encoding/json"
 	"errors"
 	"io"
 	"io/ioutil"
 	"net/http"
-)
+	"path/filepath"
+	"strings"
 
-// NewClient instantiates a Client associated with a client id, secret and environment.
-// See https://plaid.com/docs/api/#gaining-access.
-func NewClient(clientID, secret string, environment environmentURL) *Client {
-	return &Client{clientID, secret, environment, &http.Client{}}
-}
+	"github.com/BurntSushi/toml"
+)
 
-// Same as above but with additional parameter to pass http.Client. This is required
-// if you want to run the code on Google AppEngine which prohibits use of http.DefaultClient
-func NewCustomClient(clientID, secret string, environment environmentURL, httpClient *http.Client) *Client {
-	return &Client{clientID, secret, environment, httpClient}
-}
+// LatestAPIVersion holds the latest version of the Plaid API
+const LatestAPIVersion = "2018-05-22"
 
+// Client holds information required to interact with the Plaid API.
 // Note: Client is only exported for method documentation purposes.
 // Instances should only be created through the 'NewClient' function.
 //
 // See https://github.com/golang/go/issues/7823.
 type Client struct {
 	clientID    string
 	secret      string
-	environment environmentURL
+	publicKey   string
+	environment Environment
 	httpClient  *http.Client
+	_APIVersion string
```

Comment:
Why do we store this value on the client now? This library should be pinned to `2018-05-22` - the user should not be able to change this version.

Reply:
see here @r-ohan https://github.com/plaid/plaid-go/pull/40#discussion_r192576826

Reply:
⚡️